test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -901,9 +902,25 @@ public async Task WriteMetadataDocumentAsync_WorksForJsonCsdl()
 
 
             // Act - JSON
+            // We allow synchronous I/O for WriteMetadataDocumentAsync because
+            // it relies on EdmLib's CsdlWriter which is still synchronous.
+            // We should disable synchronous I/O here once CsdlWriter has an async API.
+            // See: https://github.com/OData/odata.net/issues/2684
             string payload = await this.WriteAndGetPayloadAsync(edmModel, "application/json", async omWriter =>
             {
-                await omWriter.WriteMetadataDocumentAsync();
+                try
+                {
+                    await omWriter.WriteMetadataDocumentAsync();
+                }
+                catch (Exception e)

```

Comment:
We should catch _specific_ exception here if possible

Reply:
Done. Naming suggestions would be cool too, cause I'm not sure about `ODataSynchronousIOException` :)

Reply:
@habbes ASP.NET Core throws a `System.InvalidOperationException`. I think a name like `SynchronousIOException` would do the job